src/app/components/SidebarMenu.jsx

```diff
@@ -18,17 +30,50 @@ export default class SidebarMenu extends Component {
           );
         }
       });
+
+      let originalMenuItems = _.cloneDeep(menuItems);
+
+      this.setState({menuItems, originalMenuItems});
```

Comment:
State need to be small as is possible so please don't put to state react elements.